File: 768-partition-labels/partition-labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        int n= s.length();
        List<Integer> result= new ArrayList<>();
        HashMap<Character, Integer> hm= new HashMap<>();
        for(int i=0; i< n ;i++){
            char ch=s.charAt(i);
            hm.put(ch,i);
        }

        int end=0;
        int size=0;
        for(int i=0; i<n ;i++){
            size+=1;
            end= Math.max(hm.get(s.charAt(i)),end);
            if(end==i){
                result.add(size);
                size=0;
            }

        }
        return result;

    }
}

File: 768-partition-labels/partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        map={ s[i]:i for i in range(0,len(s)) }

        end=0
        size=0
        result=[]
        for i in range(0, len(s)):
            size+=1
            end= max(end, map[s[i]])
            if(end==i):
                result.append(size)
                size=0
        return result
